core/src/main/java/com/graphhopper/routing/ev/NewCountry.java

```diff
@@ -0,0 +1,265 @@
+    JEY("Jersey"),
+    JOR("Jordan"),
+    JPN("Japan"),
+    KAZ("Kazakhstan"),
+    KEN("Kenya"),
+    KGZ("Kyrgyzstan"),
+    KHM("Cambodia"),
+    KIR("Kiribati"),
+    KNA("Saint Kitts and Nevis"),
+    KOR("South Korea"),
+    KWT("Kuwait"),
+    LAO("Laos"),
+    LBN("Lebanon"),
+    LBR("Liberia"),
+    LBY("Libya"),
+    LCA("Saint Lucia"),
+    LIE("Liechtenstein"),
+    LKA("Sri Lanka"),
+    LSO("Lesotho"),
+    LTU("Lithuania"),
+    LUX("Luxembourg"),
+    LVA("Latvia"),
+    MAR("Morocco"),
+    MCO("Monaco"),
+    MDA("Moldova"),
+    MDG("Madagascar"),
+    MDV("Maldives"),
+    MEX("Mexico"),
+    MHL("Marshall Islands"),
+    MKD("North Macedonia"),
+    MLI("Mali"),
+    MLT("Malta"),
+    MMR("Myanmar"),
+    MNE("Montenegro"),
+    MNG("Mongolia"),
+    MOZ("Mozambique"),
+    MRT("Mauritania"),
+    MSR("Montserrat"),
+    MUS("Mauritius"),
+    MWI("Malawi"),
+    MYS("Malaysia"),
+    NAM("Namibia"),
+    NER("Niger"),
+    NGA("Nigeria"),
+    NIC("Nicaragua"),
+    NIU("Niue"),
+    NLD("Netherlands"),
+    NOR("Norway"),
+    NPL("Nepal"),
+    NRU("Nauru"),
+    NZL("New Zealand"),
+    OMN("Oman"),
+    PAK("Pakistan"),
+    PAN("Panama"),
+    PCN("Pitcairn Islands"),
+    PER("Peru"),
+    PHL("Philippines"),
+    PLW("Palau"),
+    PNG("Papua New Guinea"),
+    POL("Poland"),
+    PRK("North Korea"),
+    PRT("Portugal"),
+    PRY("Paraguay"),
+    PSE("Palestinian Territories"),
+    QAT("Qatar"),
+    ROU("Romania"),
+    RUS("Russia"),
+    RWA("Rwanda"),
+    SAU("Saudi Arabia"),
+    SDN("Sudan"),
+    SEN("Senegal"),
+    SGP("Singapore"),
+    SGS("South Georgia and the South Sandwich Islands"),
+    SHN("Saint Helena, Ascension and Tristan da Cunha"),
+    SLB("Solomon Islands"),
+    SLE("Sierra Leone"),
+    SLV("El Salvador"),
+    SMR("San Marino"),
+    SOM("Somalia"),
+    SRB("Serbia"),
+    SSD("South Sudan"),
+    STP("São Tomé and Príncipe"),
+    SUR("Suriname"),
+    SVK("Slovakia"),
+    SVN("Slovenia"),
+    SWE("Sweden"),
+    SWZ("Eswatini"),
+    SYC("Seychelles"),
+    SYR("Syria"),
+    TCA("Turks and Caicos Islands"),
+    TCD("Chad"),
+    TGO("Togo"),
+    THA("Thailand"),
+    TJK("Tajikistan"),
+    TKL("Tokelau"),
+    TKM("Turkmenistan"),
+    TLS("East Timor"),
+    TON("Tonga"),
+    TTO("Trinidad and Tobago"),
+    TUN("Tunisia"),
+    TUR("Turkey"),
+    TUV("Tuvalu"),
+    TWN("Taiwan"),
+    TZA("Tanzania"),
+    UGA("Uganda"),
+    UKR("Ukraine"),
+    URY("Uruguay"),
+    USA("United States"),
+    UZB("Uzbekistan"),
+    VAT("Vatican City"),
+    VCT("Saint Vincent and the Grenadines"),
+    VEN("Venezuela"),
+    VGB("British Virgin Islands"),
+    VNM("Vietnam"),
+    VUT("Vanuatu"),
+    WSM("Samoa"),
+    XKX("Kosovo"),
+    YEM("Yemen"),
+    ZAF("South Africa"),
+    ZMB("Zambia"),
+    ZWE("Zimbabwe"),
+    MISSING("missing");
+
+    public static final String KEY = "new_country";
+
+    private final String name;
+
+    NewCountry(String name) {
```

Comment:
Can we also include the two letter country code?

Reply:
So far I used `ISO3166-1:alpha3` as country code, but no idea honestly. What are the pros and cons of the different codes? If there is some use-case for it we can of course also include other fields.

Reply:
Depends on the data. Our geojson uses alpha3 and some tags in OSM like e.g `maxspeed=RU:urban` reference the alpha2 code. I doesn't really hurt to have both representations.

Reply:
Ok I see. So should we use alpha3 for the enum constants and add alpha2 as additional field, just like name? Or maybe alpha3 as enum constant, but also as additional field?

Btw the countries properties currently contain all these fields:

```
  "id": -2186665,
  "admin_level": "2",
  "name": "Tonga",
  "local_name": "Tonga",
  "name_en": "Tonga",
  "flag": "http://upload.wikimedia.org/wikipedia/commons/9/9a/Flag_of_Tonga.svg",
  "name:af": "Tonga",
  "name:am": "ቶንጋ",
  "name:an": "Tonga",
  "name:ar": "تونجا",
  "name:az": "Tonqa",
  "name:ba": "Тонга",
  "name:be": "Тонга",
  "name:bg": "Тонга",
  "name:bn": "টোঙ্গা",
  "name:bo": "ཊོང་ག",
  "name:br": "Tonga",
  "name:bs": "Tonga",
  "name:ca": "Tonga",
  "name:ce": "Тонга",
  "name:cs": "Tonga",
  "name:cu": "Тонга",
  "name:cy": "Tonga",
  "name:da": "Tonga",
  "name:de": "Tonga",
  "name:dv": "ޓޮންގާ",
  "name:el": "Τόνγκα",
  "name:en": "Tonga",
  "name:eo": "Tongo",
  "name:es": "Tonga",
  "name:et": "Tonga",
  "name:eu": "Tonga",
  "name:fa": "تونگا",
  "name:fi": "Tonga",
  "name:fo": "Tonga",
  "name:fr": "Tonga",
  "name:fy": "Tonga",
  "name:ga": "Tonga",
  "name:gd": "Tonga",
  "name:gl": "Tonga",
  "name:gv": "Yn Tongey",
  "name:he": "טונגה",
  "name:hi": "टोंगा",
  "name:hr": "Tonga",
  "name:ht": "Tonga",
  "name:hu": "Tonga",
  "name:hy": "Տոնգա",
  "name:ia": "Tonga",
  "name:id": "Tonga",
  "name:io": "Tonga",
  "name:is": "Tonga",
  "name:it": "Tonga",
  "name:ja": "トンガ",
  "name:jv": "Tonga",
  "name:ka": "ტონგა",
  "name:kk": "Тонга",
  "name:kn": "ಟೋಂಗಾ",
  "name:ko": "통가",
  "name:ku": "Tonga",
  "name:kw": "Tonga",
  "name:la": "Tonga",
  "name:lb": "Tonga",
  "name:li": "Tonga",
  "name:lt": "Tonga",
  "name:lv": "Tonga",
  "name:mk": "Тонга",
  "name:ml": "ടോങ്ക",
  "name:mn": "Тонга",
  "name:mr": "टोंगा",
  "name:ms": "Tonga",
  "name:mt": "Tonga",
  "name:my": "တုံဂါနိုင်ငံ",
  "name:nl": "Tonga",
  "name:nn": "Tonga",
  "name:no": "Tonga",
  "name:nv": "Tʼónga",
  "name:oc": "Tònga",
  "name:or": "ଟୋଙ୍ଗା",
  "name:os": "Тонгæ",
  "name:pa": "ਟੋਂਗਾ",
  "name:pl": "Tonga",
  "name:pt": "Tonga",
  "name:qu": "Tunqa",
  "name:rm": "Tonga",
  "name:ro": "Tonga",
  "name:ru": "Тонга",
  "name:rw": "Tonga",
  "name:sa": "टोंगा",
  "name:se": "Tonga",
  "name:sh": "Tonga",
  "name:sk": "Tonga",
  "name:sl": "Tonga",
  "name:sm": "Toga",
  "name:sq": "Tonga",
  "name:sr": "Тонга",
  "name:su": "Tonga",
  "name:sv": "Tonga",
  "name:sw": "Tonga",
  "name:ta": "தொங்கா",
  "name:tg": "Тонга",
  "name:th": "ประเทศตองกา",
  "name:tl": "Tonga",
  "name:to": "Tonga",
  "name:tr": "Tonga",
  "name:ts": "Tonga",
  "name:tw": "Tonga",
  "name:ty": "To’a",
  "name:ug": "Ton’ga",
  "name:uk": "Тонга",
  "name:ur": "ٹونگا",
  "name:uz": "Tonga",
  "name:vi": "Tonga",
  "name:vo": "Tonguäns",
  "name:wo": "Tonga",
  "name:yo": "Tóngà",
  "name:zh": "東加",
  "boundary": "administrative",
  "maritime": "yes",
  "name:ace": "Tonga",
  "name:als": "Tonga",
  "name:arz": "تونجا",
  "name:ast": "Tonga",
  "name:bcl": "Tonga",
  "name:bjn": "Tonga",
  "name:bpy": "টঙ্গা",
  "name:bxr": "Тонга",
  "name:ceb": "Tonga",
  "name:chy": "Tonga",
  "name:ckb": "تۆنگا",
  "name:crh": "Tonga",
  "name:diq": "Tonga",
  "name:ext": "Tonga",
  "name:frp": "Tonga",
  "name:hak": "Tûng-kâ",
  "name:hif": "Tonga",
  "name:ilo": "Tonga",
  "name:lij": "Tonga",
  "name:lmo": "Tonga",
  "name:mrj": "Тонга",
  "name:nah": "Tonga",
  "name:nan": "Tonga",
  "name:nds": "Tonga",
  "name:pam": "Tonga",
  "name:pih": "Tonga",
  "name:pms": "Tònga",
  "name:pnb": "ٹونگا",
  "name:sah": "Тонга",
  "name:scn": "Tonga",
  "name:sco": "Tonga",
  "name:smn": "Tonga",
  "name:sms": "Tonga",
  "name:szl": "Tůnga",
  "name:tpi": "Tonga",
  "name:tzl": "Tonga",
  "name:war": "Tonga",
  "name:xal": "Тонһлмудин Нутг",
  "name:yue": "湯加",
  "timezone": "Pacific/Tongatapu",
  "way_area": "3.34815",
  "wikidata": "Q678",
  "ISO3166-1": "TO",
  "wikipedia": "en:Tonga",
  "name:simple": "Tonga",
  "driving_side": "left",
  "name:bat-smg": "Tuonga",
  "name:fiu-vro": "Tonga",
  "name:be-tarask": "Тонга",
  "is_in:continent": "Oceania",
  "ISO3166-1:alpha2": "TO",
  "ISO3166-1:alpha3": "TON",
  "official_name:cs": "Království Tonga",
  "official_name:de": "Königreich Tonga",
  "official_name:en": "Kingdom of Tonga",
  "official_name:es": "Reino de Tonga",
  "official_name:ru": "Королевство Тонга",
  "ISO3166-1:numeric": "776"
}
```
Since we want to move this into the bundle we should probably get rid of what we do not use (just in case this actually reduces the filesize).

Reply:
nv-i18n handles this in a similar way: https://github.com/TakahikoKawasaki/nv-i18n/blob/master/src/main/java/com/neovisionaries/i18n/CountryCode.java

One code representation as the name of the constant and the other one as a regular value.

Reply:
Ok, so we will include the alpha2 and alpha3 codes, and a readable name I would say. Which name should we keep? `official_name:en`, `name:simple`, `name`, `local_name`, `name_en`, `name:en`?  I'm a bit lost here... What about `ISO3166-1:numeric`? Some other interesting fields are: `driving_side`, `is_in:contintent`, `timezone`? I just checked, not including any properties the `countries.geojson` file will be 1.3MB vs 2.7MB currently. So not too important, but getting rid of the many country name translations could be worth it.

Reply:
> Since we want to move this into the bundle

What do you mean here? Isn't the file already in the jar? And I do not think that the properties matter much size-wise, especially since everything is compressed in the jar.

Reply:
No, it's in core/files/spatialrules currently. The compressed size of the countries file is 900kb currently and 450kb if no properties were included. For a comparison the core jar is 1.1MB currently.

I was asking myself two questions: Can we get rid of some properties we aren't using anyways (just to optimize jar size)? And which fields should we include in the country enum, and which ones might be useful?

Reply:
Most of these tags don't work for countries with oversea territories like France: https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=2202162&class=boundary

I'd go with these:

* name:en
* ISO3166-1:alpha2
* ISO3166-1:alpha3

ISO3166-1:numeric won't hurt but i don't see a usecase for it within GraphHopper.

Reply:
> No, it's in core/files/spatialrules currently. The compressed size of the countries file is 900kb currently and 450kb if no properties were included. For a comparison the core jar is 1.1MB currently.

We could probably reduce the file size further if we simplified all the non-shared borders. We don't need to maintain full precision along coastlines or the shape of individual islands in an archipelago. But I'm afraid this will be a manual process, or at least I don't know of any tools or algorithms to automate this. And it's probably not worth the time as the current filesize without additional properties is good enough IMHO.

Reply:
> I'd go with these:

Ok thanks and good to know. You mean things like timezone and continent don't work for oversea territories?

> And it's probably not worth the time as the current filesize without additional properties is good enough IMHO.

Yes, I'm not too worried about the file size and we should definitely not compromise on the border precision or something. I just thought removing the many translated country names would be a quick win.

Reply:
> No, it's in core/files/spatialrules currently. 

Ah, ok. Thanks. Strange, always thought we have it in the bundle already. 

> So should we use alpha3 for the enum constants and add alpha2 as additional field, just like name? 

This sounds good to me :+1: 

> driving_side

This is indeed interesting but not sure if we should attach it to the enum or if it belongs to the country rule.

Reply:
Timezones are a completely different caliber:

![Timezone map](https://upload.wikimedia.org/wikipedia/commons/8/88/World_Time_Zones_Map.png)

For example:

* USA: 6 timezones
* Russia: 11(!) timezones

Reply:
Ok, sure. Russia actually does not even have a timezone field (probably for this reason).

Reply:
Should we keep `alpha2`, `alpha3` `name:en`, `ISO3166-1:numeric`, `flag` and `wikidata` in `countries.geojson` and remove everything else? The last three wouldn't be used by default but I thought they might be useful to somebody?

Reply:
I did this